### DISCUSSION TITLE
Add zoom page setting for printing

### DISCUSSION THREAD

**Discussion by TajangSec** [sumatrapdfreader/sumatrapdf#4125] [Ideas]
English：When I print some book PDFs, they are usually not as big as A4 paper. sumatraPDF uses the Windows default print window, which has no zoom function. So after printing, the books on the paper are very small. The following figure is the SumatraPDF printing results.
_Chinese：当我打印一些书籍PDF时，它们通常没有A4纸那么大，sumatraPDF使用的Windows默认打印窗口，这个窗口是没有缩放功能的。所以打印后，纸张上的书籍非常小。下图是SumatraPDF打印结果。_
![1](https://github.com/sumatrapdfreader/sumatrapdf/assets/63721558/9496241f-06bc-4806-8d90-e27ffdb106fc)

English：And Google Chrome has zoom function when printing PDF.
_Chinese：而Google Chrome打印PDF时有缩放页面功能。_
![PixPin_2024-03-08_19-04-33](https://github.com/sumatrapdfreader/sumatrapdf/assets/63721558/8b5d71df-c252-4857-941e-9703e900deac)
![2](https://github.com/sumatrapdfreader/sumatrapdf/assets/63721558/186184fb-04c3-4e84-bb6b-448b2a9b9017)

English：Is it possible to add zoom function in SumatraPDF?
_Chinese：请问能否在在SumatraPDF中加入缩放功能？_

**Comment by GitHubRulesOK** [comment]
SumatraPDF printing is as simple as can be so the whole page will be sent to the printer as an image with the documents native white space there is only one way to trim that whitespace and that is by manual selection (the one that can be used for copy and zoom by steps).
Thus to print only a zoomable area is to use print selection thus only good for one GUI page at a time.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/69c99159-f3e0-4522-bcad-74e32076eec2)


The only other related option is to enlarge the normal page boundary to fit the paper print area, (advanced settings **PrintScale = fit**)
 and that is possibly what you have there
 the page "margins" + the printer default page gripper + set margins 
So potentially a box in a box in a box in a box.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/53d86c05-f132-465e-a010-01ce2f17774b)

**Reply by TajangSec** [reply]
I'm Windows 11, and in my printer settings, I don't have any of these options, only a choice of pages, and copies. Only color can be selected in the advanced settings. And there is no preview, but Google Chrome's print window can be zoomed and previewed, so I am now reading the file with SumatraPDF and printing using Google Chrome!

**Reply by GitHubRulesOK** [reply]
For printing PDF best there is either Adobe GUI Apps or for command line printing the sister product to the Artifex MuPDF core in SumatraPDF which would be Artifex GhostScript

Microsoft include Acrobat view and print in MS Edge so no need for Chrome. You can even sent the command to open Edge viewer from sumatraPDF for a page of a PDF (or an ePub  for view, but it cannot currently print ePubs)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/93ee9efe-ec97-4e78-87e7-96b6e287d6c6)
